### ISSUE TITLE
Unable to adjust the zones in the .jar or the .deb versions.

### ISSUE THREAD

**Issue by user90919293949596979899** [gtasa-savegame-editor/gtasa-savegame-editor#97]
I can't change the zones in the editor for some reason.

**Comment by lfuelling** [comment]
The zone text fields stay diabled and without value regardless of the zone selected and the "gang wars" setting.

<img width="1012" alt="Screenshot 2021-06-19 at 17 13 05" src="https://user-images.githubusercontent.com/5144843/122646853-9e022a00-d121-11eb-9233-27ce9551ef22.png">

**Comment by lfuelling** [comment]
Unfortunately, I didn't write that (or most of) the code, I just added some macOS-specific features, started work on support for garages and fixed the project structure, build process and documentation. 

I tried multiple times to understand what that code is supposed to do but I always get lost on the way and end up with no time left. Since you mentioned that using the debugger works, I'd suspect some sort of race condition or as you've mentioned an issue with reading inputs. 
I don't really understand that part of the code and haven't properly coded Java in almost three years now, so I'm sorry but I can't give you any more info on the specifics of the Zone tab, than:

From what I understand, users are supposed to select an area of the map, which will then result in the zones on the map being highlighted, so they can set the gangs that are active in that zone.
The `handleEvent` method you mentioned probably sets the disabled state of the text fields, they should be disabled when nothing is selected, but I'm not sure if that event even is fired correctly when an area is selected.

**Comment by lfuelling** [comment]
The website linked at the repository start page is the original website, @paul35621 is the original developer.
I just took over maintainership hoping I could fix what kept it from working for the macOS version, add some quality-of-life improvements, and maybe even get some outside contributions that fix other issues by hosting it in a more accessible manner on GitHub.

You can try reaching out to him using email if that's still available on his website, but I can't guarantee that he'll respond. 

In any case, even if you just clean up the code a bit to make it more readable, any contribution back is more than welcome, maybe it will help someone else who tries to tackle this issue.

**Comment by lfuelling** [comment]
No, the code was previously only available on pauls website (linked at the top of this repo) as a zip without any way to contribute changes, so I wrote an email asking if he would be okay with putting the project on GitHub so I could fix the issues I have and maybe develop further features, to which he replied that I should then also take over maintainership (ie. being one of the people able to merge commits, reply to issues, etc.) because he doesn't have the time for all that.
As far as I know Paul created his GitHub account only because I wanted to add him to the group.

As I said, if you manage to get further even a little bit, please consider creating a PR with your changes so other people (maybe me should I suddenly get motivated again) can continue your work.

**Comment by lfuelling** [comment]
Please try [v3.3-rc.30](https://github.com/gtasa-savegame-editor/gtasa-savegame-editor/releases/tag/v3.3-rc.30), it should work now.

**Comment by rain1** [comment]
Would it be possible to have it compiled for older than java 23?

> version of the Java Runtime (class file version 67.0), this version of the Java Runtime only recognizes class file versions up to 55.0

**Comment by lfuelling** [comment]
I'm not sure about any dependencies, but it might work. Have you already tried changing the version in the pom on your end?

**Comment by rain1** [comment]
Ok unfortunately there is dependency darklaf-core-3.1.1.jar that needs java 17 at minimum. My initial idea was that bumping java version higher than necessary will unnecessarily cut off players with older java versions.

**Comment by lfuelling** [comment]
You're right, unfortunately Java 9 is the lowest I was able to go (by switching from darklaf to flatlaf); You can try [v3.3-rc.31](https://github.com/gtasa-savegame-editor/gtasa-savegame-editor/releases/tag/v3.3-rc.31) once it's done building.

**Comment by rain1** [comment]
Thanks. I can confirm that jar works and zones can be edited now.
